frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -63,9 +64,7 @@ function FileManagerMenu:initGesListener()
 end
 
 function FileManagerMenu:setUpdateItemTable()
-    for _, widget in pairs(self.registered_widgets) do
-        widget:addToMainMenu(self.tab_item_table)
-    end
+    PluginLoader:addToMenu(self.registered_widgets, self.tab_item_table)
```

Comment:
As you already mentioned, it's a little bit confusing to things for non-plugin related stuff in PluginLoader. Aside from this, we are adding magic to alter the expected behavior in all plugins' addToMainMenu methods, specifically the `table.insert(tab_item_table.plugins` call.

If we are going to prevent plugins from having full control over their positioning in the menu, we should replace those `addToMainMenu` methods with methods with higher abstractions.

For example, we can let all plugins register menu entries with pluginloader instead, then in `setUpdateItemTable` retrieve the plugin menu from pluginloader and merge it to the rest of the menus.